src/providers/auth-login-redirect-token-auth-clear.js

```diff
@@ -0,0 +1,43 @@
+(function (angular) {
+    'use strict';
+
+    var module = angular.module('angular-token-auth-login-redirect-token-auth-clear', [
+        // For $route
+        'ngRoute',
+        'angular-token-auth-login-redirect-token-auth-settings'
+    ]);
+
+    module.service('authLoginRedirect.onTokenAuthClear', [
+        '$route',
+        '$location',
+        'authLoginRedirect.getTokenAuthSettings',
+        function ($route, $location, getTokenAuthSettings) {
+
+            this.handler = angular.bind(this, function onTokenAuthClear () {
+                if (!this.routeIsAnonymous($route.current.$$route)) {
+                    this.redirect();
+                }
+            });
+
+            this.routeIsAnonymous = function (route) {
+                // By default, all routes should be anonymous.
+                var routeIsAnonymous = true;
+                if (route && route.anonymous === false) {
+                    routeIsAnonymous = false;
+                }
+                return routeIsAnonymous;
```

Comment:
I think being explicit here is better: this function returns a boolean. Also the `route &&` part isn't checking if `route` is defined, it's just catching the situation where it's not so it doesn't TypeError on `route.anonymous`.